app/experimenter/jetstream/client.py

```diff
@@ -57,9 +59,25 @@ def get_results_metrics_map(data, primary_outcome_slugs, secondary_outcome_slugs
         )
     )
 
+    try:
+        valid_metrics = set(
+            chain.from_iterable(
+                [
+                    experiment_metadata["outcomes"][slug]["metrics"]
+                    + experiment_metadata["outcomes"][slug]["default_metrics"]
+                    for slug in experiment_metadata["outcomes"]
+                ]
+            )
+        )
+    except (TypeError, KeyError):
+        # could not retrieve metadata from Jetstream
+        valid_metrics = None
```

Comment:
Yea, thinking about this a little more we should be able to make this less convoluted. The keys we are accessing should always exist as long as we can retrieve metadata from Jetstream. If we can't retrieve from Jetstream, we'll fail before getting here, so that's covered. If somehow we got here and the expected keys don't exist in the metadata, we should probably fail anyway because it means the metadata looks different than we expect.

I was also thinking that we would be OK passing through without validating against Jetstream metadata if the experiment_metadata was not available for some reason, but I'm not sure that's the case. The examples I've tried would all be fine doing this, but I don't know that it's a safe assumption. Better to just use the metadata if we have it, and fail if not (and like I mentioned earlier, we should likely be failing earlier if we don't have it anyway).

Reply:
Yeah exactly, making assumptions about data and then completely falling over and erroring out to sentry is a completely valid strategy, especially in background tasks like this, so I'm totally good with that 👍

Reply:
Question: we have this function earlier in `client.py`:

```
def load_data_from_gcs(path):
    if default_storage.exists(path):
        return json.loads(default_storage.open(path).read())
```

In the case that `default_storage.exists(path)` returns `False`, the metadata will be `None`, but we don't currently fail. Should we a) handle this in the code block referenced above, or b) remove the `default_storage.exists(path)` check and let an exception happen (leave the check but explicitly raise an exception in the else case)?

For now, I went with (a), but happy to update or discuss if there's something I missed.